### PR TITLE
Add check-node-version

### LINKED ISSUE #5508: Documentation enhancement for getting started section
# Documentation enhancement request for getting started section

## Example name
Getting started

## Describe the bug
Users who do not have the node version 7.0.0 and above on their local machine are shown with the error `Object.values is not a function` in the terminal. The very new user first command to run the `next.js`application resultants in error.

## To Reproduce
Steps to reproduce the behavior, please provide code snippets or a repository:
1. Go to `getting started` section.
2. Follow the steps mentioned
3. Execute `npm run dev`
4. If the user's node version is not 7 and above in his/her local machine terminal throws `Object.value` error.

## Expected behaviour
Shall we have a `note` saying the node version should be 7 and above in the installation steps

## System information
 - OS:  macOS
 - Version of Next.js: [e.g. 7.0.2]

## Additional context

If the reviewers agree for this enhancement, I would like to contribute to this change.

### PR DESCRIPTION
**Pull request by shuding**
Fixes #5508

### CODE REVIEW

**Review by HaNdTriX — COMMENTED** [review]
We might want to use the engines field from the `package.json` for that.

* https://yarnpkg.com/lang/en/docs/package-json/
* https://docs.npmjs.com/files/package.json#engines

File: packages/next/server/lib/check-node-version.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { printAndExit } from './utils'
+
+const NODE_VERSION = process.version.match(/^v(\d+)/)[1]
+
+if (NODE_VERSION < 7) {
+  printAndExit(`> You're using Node.js with version less than 7 which is not supported. Please upgrade to Node.js >= 7.`)
```
Comment (timneutkens) [inline packages/next/server/lib/check-node-version.js]: Node.js 7.5+ is supported

File: packages/next/build/webpack.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 // @flow
+
+// Check node version before including other dependencies
+import '../server/lib/check-node-version'
```
Comment (timneutkens) [inline packages/next/build/webpack.js]: 🤔 This should probably be used in `next-server/server/next-server.js` instead.

**Comment by radeno** [comment]
Agree with @HaNdTriX to not polluting code with version checking. Just define it in package.json
https://docs.npmjs.com/files/package.json#engines

**Comment by timneutkens** [comment]
Even if someone force installs it / npm only shows a warning we should show a clear message instead of erroring out with a random stacktrace

**Comment by radeno** [comment]
@timneutkens i agree, but package.json is first place where people will search dependencies. even engine dependencies. like in Gemfile in Ruby

**Comment by radeno** [comment]
@timneutkens should we check just node or also package managers? minimum version for node is 7.5 an up, right?

**Comment by timneutkens** [comment]
Closing this in favor of #5724

**Pull request closed without merging**